### PR TITLE
BACKLOG-7817 - Location prompt in report viewer: custom prompting dialog

### PR DESCRIPTION
**Pull request by EgorZhuk**
Just for Wingman. Please don't merge.

### CODE REVIEW

File: user-console/source/org/pentaho/mantle/client/commands/AdhocRunInBackgroundCommand.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  @Override
+  protected void performOperation( boolean feedback ) {
+
+    RequestBuilder treeRequestBuilder = createTreeRequest();
+
+    try {
+      treeRequestBuilder.sendRequest( null, new RequestCallback() {
+
+        public void onError( Request request, Throwable exception ) {
+          MessageDialogBox dialogBox =
```
Comment (tmorgner) [inline user-console/source/org/pentaho/mantle/client/commands/AdhocRunInBackgroundCommand.java]: This error handling code (and its second variation with a response-status) is replicated multiple times. Condense it into a single method?

File: user-console/source/org/pentaho/mantle/client/commands/AdhocRunInBackgroundCommand.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+  public void setJobId( String jobId ) {
+    this.jobId = jobId;
+  }
+
+  public String getJobId() {
+    return jobId;
+  }
+
+  @Override public String getOutputLocationPath() {
```
Comment (tmorgner) [inline user-console/source/org/pentaho/mantle/client/commands/AdhocRunInBackgroundCommand.java]: You are overriding plain getters from the parent class. Is that necessary, given that the parent has no logic in either getter or setter? You should be able to just use the base-versions instead. Removes a few lines of duplicate code.

File: user-console/source/org/pentaho/mantle/client/commands/AdhocRunInBackgroundCommand.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        public void onError( Request request, Throwable exception ) {
+          MessageDialogBox dialogBox =
+            new MessageDialogBox( Messages.getString( "error" ), exception.toString(), false, false,
+              true );
+          dialogBox.center();
+        }
+
+        public void onResponseReceived( Request request, Response response ) {
+          if ( response.getStatusCode() == Response.SC_OK ) {
+            String folderId = null;
```
Comment (tmorgner) [inline user-console/source/org/pentaho/mantle/client/commands/AdhocRunInBackgroundCommand.java]: Could we extract this block into a "parseFolderId(Response response)" method? Makes the block a lot easier to read. Along with moving the error handling into extra methods (see above), this then becomes a trivially simple to read method. I love trivially simple - it matches my brain :)

**Comment by EgorZhuk** [comment]
@tmorgner ok, I will refactor it. Not sure for getters and setters - there are some problems with GWT, but I will try to understand what is going on.

**Comment by tmorgner** [comment]
If the getter and setter cause problems, then screw that part and play it safe. But we can at least make the main method a hell of a lot more readable via simple refactorings.

**Comment by EgorZhuk** [comment]
@tmorgner in fact we needed only solutionPath inside child class

**Pull request merged**